openedx/adg/admin_dashboard/applications/helpers.py

```diff
@@ -0,0 +1,103 @@
+
+    if not (user.is_superuser or is_adg_admin(user)):
+        submitted_applications = submitted_applications.get_applications_for_business_unit_admin(
+            user.groups.first()
+        )
+
+    return submitted_applications
+
+
+def get_new_applications_count_and_percentage_change(submitted_applications):
+    """
+    Calculates the number of new applications in the current month and the percentage change between the number of
+    submitted applications in the current and previous month.
+
+    Arguments:
+        submitted_applications (QuerySet): queryset containing submitted applications
+
+    Returns:
+         new_applications_count (int): total number of submitted application in the current month
+         percentage (int): percentage change between submitted applications in the current and previous month
+    """
+    today = datetime.today()
+    start_of_month = datetime(today.year, today.month, 1)
+    start_of_previous_month = datetime(today.year, today.month - 1, 1)
+
+    new_applications_count = submitted_applications.get_submitted_applications_count_in_date_range(
+        start_of_month, today
+    )
+
+    applications_count_in_last_month = submitted_applications.get_submitted_applications_count_in_date_range(
+        start_of_previous_month, start_of_month - timedelta(days=1)
+    )
+
+    if applications_count_in_last_month == 0:
+        percentage = new_applications_count * 100
+    else:
+        percentage = int(
+            (new_applications_count - applications_count_in_last_month) / applications_count_in_last_month * 100
+        )
+
+    return new_applications_count, percentage
+
+
+def get_weekly_application_submissions(submitted_applications, number_of_weeks):
+    """
+    Calculates the number of submitted applications for the past few weeks, sent as argument, including the current
+    week.
+
+    Arguments:
+        submitted_applications (QuerySet): queryset containing submitted applications
+        number_of_weeks (int): number of weeks for which weekly submissions need to be calculated
+
+    Returns:
+        JSON: a json object containing two lists: one for dates of the past 12 weeks and the other for number of
+        application submissions in those weeks
+    """
+    today = datetime.today()
+    start_of_week = today - timedelta(days=today.weekday())
+
+    submissions_list = [0] * number_of_weeks
+    date_list = [''] * number_of_weeks
+    start_date = start_of_week
+    end_date = today
+
+    for index in range(number_of_weeks - 1, -1, -1):
+        filtered_applications_count = submitted_applications.get_submitted_applications_count_in_date_range(
+            start_date, end_date
+        )
+
+        submissions_list[index] = filtered_applications_count
+        date_list[index] = start_date.strftime(WEEKLY_SUBMISSIONS_GRAPH_DATE_FORMAT)
+
+        end_date = start_date - timedelta(days=1)
+        start_date = start_date - timedelta(days=7)
```

Comment:
Can we convert this to dictionary instead of using the lists?

Reply:
While making the graph we need two separate lists: one for labels and the other for data and to make dict here and then convert to array in JS will increase the computation